### ISSUE TITLE
Any explanation for the little performance gain?

### ISSUE THREAD

**Issue by hy9be** [austinksmith/Hamsters.js#7]
I was trying to see how much performance gain I should expect from multi-thread my task. So I tried the test on the [official site](http://www.hamsters.io/performance/#). But surprisingly, the result indicated that there was very little performance gain even on the 25.6 million cases.

I am aware of the Amdahls Law and Gustafson Law, but still cannot figure out a reasonable explanation. Anyone have some ideas?

Screenshot for my result FYI:
<img width="1393" alt="screen shot 2016-04-02 at 11 31 31 am" src="https://cloud.githubusercontent.com/assets/3392907/14227222/8317d9b2-f8c6-11e5-9fb1-5cb9602dc157.png">

**Comment by austinksmith** [comment]
Hey there, sorry i didn't see your issue request until now. What browser are you testing with?

**Comment by hy9be** [comment]
Thanks for the information @austinksmith !

I am on Chrome 49, the machine is a 2013 Macbook pro, with
- Processor: 2.7 GHz Intel Core i7
- Memory: 16 GB 1600 MHz DDR3

Given that the navigator.hardwareConcurrency is supported and the hardware is sufficient to leverage web worker. Do you mean the case on the official site is actually not the best case to show the performance improvements? Or these overhead actually occurs to all cases.

Thanks!

**Comment by austinksmith** [comment]
If you wanted to present your use case I can write out the function that will make it run at its peak, the last few releases have been focused on adding features and fixing bugs and performance optimizations have fallen a bit to the back burner while I work on getting all the kinks ironed out again. The use case on the example site is a very simplistic test that doesn't represent a real world example because there is overhead from measuring the performance itself. Your machine specs are pretty on par to mine and I've seen performance gains of over 400% in the past using the benchmark. It's also possible that Chrome has changed something to make the performance gains not as apparent in the most recent releases in which case I'll have to optimize it again in the next release. The gains you will see aren't as immediately apparent from the test mainly because you're doing a very simplistic multiplication operation that doesn't neccesarily stress each core to its maximum and after a certain point you will see diminishing returns. 

The ideal use case would be to thread as many of your functions as possible (where needed) because not only can you gain more throughput with a single function across multiple cores in parallel but you'll gain a ton of responsiveness simply by exploiting the extra threads to do multiple independent functions across individual cores that would be Massively Async/Concurrency.

**Comment by hy9be** [comment]
Get it. I will go ahead to use the library in our own case and see whether the performance improvement is more significant. And I will also see how it goes with older Chrome / older version of the library.

Thanks!

**Comment by austinksmith** [comment]
Be sure to checkout the latest dev branch version, turns out performance was being hampered because there was a bug in the memoization features that was causing indexedDB to be called after each function even if you had caching disabled in the library. The perf example was updated really early this morning to reflect those changes and the performance gains are back. Sorry for the bug.

**Comment by hy9be** [comment]
Good to know. Thanks a lot! @austinksmith

**Comment by austinksmith** [comment]
Just updated dev again today, make sure to use the non minified version for now

**Comment by austinksmith** [comment]
Closing this issue, turns out the performance example I wrote on the website was somewhat inaccurate and was measuring the time it took to execute the dom updates as well. It has been updated to eliminate the issue as best as possible and performance gains are much more apparent now, even in safari which previously only lost performance. Please make a new issue request if you have any more concerns, thanks.

**Comment by austinksmith** [comment]
@hy9be I know this is an old ticket however, I am wondering if you could try the latest version of the library today and let me know your thoughts on the performance now? Try using version 5.3.9 or version 5.4.0, 5.3.9 will be a little faster but will consume more memory during operation
